rapid7_insightvm_cloud/icon_rapid7_insightvm_cloud/util/api.py

```diff
@@ -0,0 +1,73 @@
+import json
+
+from requests import request
+from logging import Logger
+
+import insightconnect_plugin_runtime
+from insightconnect_plugin_runtime.exceptions import PluginException
+from requests.exceptions import HTTPError
+
+
+class IVM_Cloud:
+    def __init__(self, token: str, logger: Logger, url: str):
+        self.logger = logger
+        self.token = token
+        self.base_url = url
+
+    def call_api(self, path: str, request_type: str, params: dict = None, body: dict = None):
+        if params is None:
+            params = {}
+        if body is None:
+            body = {}
+
+        api_url = self.base_url + path
+
+        headers = {"x-api-key": self.token, "content-type": "application/json"}
+
+        try:
+            response = request(
+                request_type,
+                self.base_url + path,
+                params=insightconnect_plugin_runtime.helper.clean(params),
+                headers=headers,
+                data=json.dumps(body),
+            )
+            if response.status_code not in [200, 201, 202]:
+                raise PluginException(
+                    cause=f"Failed to get a valid response from InsightVM at endpoint {api_url}",
+                    assistance=f"Response was {response.request.body}",
+                    data=response.status_code,
+                )
+            if response.text == "":
+                return response.status_code
+            else:
+                return response.json()
```

Comment:
There are two different return types here which is not a good practice. It may be a good idea to return a tuple here instead, maybe (status_code, response_json) so that the return is consistent across the function usage. This will also allow you to add a return type hint to the function to indicate a consistent return type.